### PR TITLE
Avoid restarting workers when nanny waits

### PR DESCRIPTION
**Pull request by mrocklin**
Fixes https://github.com/dask/distributed/issues/1770

### CODE REVIEW

File: distributed/nanny.py [hunk 1]
```diff
@@ -356,14 +356,17 @@ def start(self):
                 yield self.process.start()
                 if self.status == 'starting':
                     yield gen.with_timeout(timedelta(seconds=5),
```
Comment (jakirkham) [inline distributed/nanny.py]: Could we add this timeout to the config file?

File: distributed/nanny.py [hunk 2]
```diff
@@ -356,14 +356,17 @@ def start(self):
...
-                                           self._wait_until_running())
+                                           self._wait_until_started())
             except gen.TimeoutError:
                 logger.info("Failed to start worker process.  Restarting")
                 yield gen.with_timeout(timedelta(seconds=1),
```
Comment (jakirkham) [inline distributed/nanny.py]: Not sure if it is needed, but it might be nice if this was configurable too.

Comment (mrocklin) [reply distributed/nanny.py]: Yeah, ideally all of the timeouts everywhere would be configurable.  I'm not yet sure how best to do this.

File: distributed/nanny.py (hunk 1 above)
Comment (jakirkham) [reply distributed/nanny.py]: Pointing this one out as people seem to already be hacking the code to change this anyways as illustrated by [this comment]( https://github.com/dask/distributed/issues/1770#issuecomment-368921388 ) and [this one]( https://github.com/dask/distributed/issues/1770#issuecomment-369013195 ). Even gave it a try myself while doing some debugging earlier. Of course naming is hard, maybe `worker_startup_wait_time`?

**Comment by mrocklin** [comment]
This timeout shouldn't be necessary.  There are other things going on that
are necessitating the hacking around.

File: distributed/nanny.py (hunk 1 above)
Comment (mrocklin) [reply distributed/nanny.py]: This is no longer the same timeout.  This is the time for the process to start and send a message back, not the time it takes to connect (which is what is was for the last version)

**Comment by jakirkham** [comment]
Hmm...so what is the motivation for the timeout?

**Comment by mrocklin** [comment]
Very occasionally in our tests worker processes don't start up.  I don't know why.  I spent a couple days looking at this a few weeks ago and could come up with anything.

**Comment by jakirkham** [comment]
Is there an issue or PR that I can read about this?

**Comment by mrocklin** [comment]
Unfortunately it appears that I didn't take very good notes.  The change
was made here:

https://github.com/dask/distributed/pull/1723

And something similar around the same time was discussed here:

https://github.com/dask/distributed/pull/1651#issuecomment-361468976

Although thinking more about that now I think that that might be a separate
issue, which is still live if you rename

distributed/tests/test_failed_workers.py
to distributed/tests/test_worker_failed.py :)

**Pull request merged**